### PR TITLE
Stream Date Filter

### PR DESCRIPTION
**Pull request by s-tyshchenko**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [x] When resolving a specific issue, it's referenced in the PR's description (e.g. `Fix #xxx[,#xxx]`, where "xxx" is the Github issue number)
- [x] All tests are passing
- [ ] New/updated tests are included
- [x] Changelog was modified

If adding a **new feature**, the PR's description includes:
- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
Issue: https://github.com/humhub/humhub/issues/5499
Screenshot:

Снимок экрана 2022-03-02 в 01 40 33

### CODE REVIEW

**Comment by ArchBlood** [comment]
In normal development that I've been apart of it is known to use the created year for new files, does HumHub not do this @luke-?

File: protected/humhub/modules/stream/models/filters/DateStreamFilter.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) 2018 HumHub GmbH & Co. KG
```
Comment (luke-) [inline protected/humhub/modules/stream/models/filters/DateStreamFilter.php:4]: @s-tyshchenko Can you please use the current year for new files?

File: protected/humhub/modules/stream/widgets/WallStreamFilterNavigation.php [hunk 2]
```diff
@@ -59,15 +64,12 @@
...
+     * Panel columns
      */
-    const PANEL_POSITION_LEFT = 0;
-
-    /**
-     * Left panel position
-     */
-    const PANEL_POSITION_CENTER = 1;
-    const PANEL_POSITION_RIGHT = 2;
+    const PANEL_COLUMN_1= 0;
```
Comment (luke-) [inline protected/humhub/modules/stream/widgets/WallStreamFilterNavigation.php:69]: @s-tyshchenko I am a bit concerned that modules may use these constants.

Can you please check if modules from https://github.com/humhub or https://github.com/humhub-contrib access these constants?

If they are not used, however, we need to mark that as a Breaking Change in the doc.

**Review by luke- — CHANGES_REQUESTED** [review]
@s-tyshchenko Can you please add some tests for it?

File: protected/humhub/modules/stream/widgets/WallStreamFilterNavigation.php (hunk 2 above)
Comment (s-tyshchenko) [reply protected/humhub/modules/stream/widgets/WallStreamFilterNavigation.php:69]: No, I haven't found any usage of these constants.

**Comment by luke-** [comment]
@yurabakhtin Can you please review

**Comment by yurabakhtin** [comment]
@luke- I have reviewed, but I could not fix tests even after 3 commits, however on my local server it works well without errors.

**Comment by yurabakhtin** [comment]
> @yurabakhtin The problem seems to be related to different date formats. I have fixed/sticked the date format here and the tests and filters run correctly.
> 
> https://github.com/humhub/humhub/blob/06c1c55ef9d96324af44e2d8e12bbe10e8e6f8d2/protected/humhub/modules/ui/filter/widgets/DatePickerFilterInput.php#L60
> 
> Can you please try again to use the respective language formats correctly? Maybe you can also have a look how the Birthday field was solved.

@luke- I tried a solution in commits https://github.com/humhub/humhub/pull/5580/commits/37a21d7fcf9163c6c8ae7e619dc9cca7200a0616, https://github.com/humhub/humhub/pull/5580/commits/0312b3a4e8fbf1134b88491394de7cc5f78f48b3, but it still has errors on the github test server, however on my local server it works well, ok I will try to find better solution later...

**Comment by luke-** [comment]
@yurabakhtin Ok, thanks. Unfortunately, the date formats are always very tough. Have you tried different locales on your local test?

**Review by luke- — APPROVED** [review]

**Comment by luke-** [comment]
@yurabakhtin Thanks looks nice. Do you see any possibility to show an error if FilterFrom is after FilterTo?

**Comment by luke-** [comment]
@yurabakhtin Can you please take a look into the tests?

**Comment by yurabakhtin** [comment]
@luke- 

> Do you see any possibility to show an error if FilterFrom is after FilterTo?

Commit https://github.com/humhub/humhub/pull/5580/commits/f913b229a121d56ba0d6ecd3a6cfbdc3243a62c1:

filter_erros

**Comment by yurabakhtin** [comment]
@luke- 

> Can you please take a look into the tests?

I could not reproduce the error, and my latest commit doesn't have the errors.

**Pull request merged**